huntsman/observatory.py

```diff
@@ -386,6 +386,78 @@ def take_evening_flats(self,
                 self.logger.debug('Waiting for dark-field image')
                 time.sleep(1)
 
+    def take_dark_fields(self,
+                         wait_interval=5,
+                         camera_list=None,
+                         exptimes_list=[],
+                         n_darks=10,
+                         *args, **kwargs
+                         ):
+        """Take dark fields
+
+        Args:
+            wait_interval (float, optional): Time in between dark exposures
+            camera_list (list, optional): List of cameras to use for darks
+            exptimes_list (list, optional): List of exposure times for darks
+            n_darks (int, optional): Number of darks to be taken per exptime
+            *args (TYPE): Description
+            **kwargs (TYPE): Description
+
+        """
+
+        image_dir = self.config['directories']['images']
+
+        dark_obs = self._create_dark_observation(n_darks, exptimes_list)
+        exptimes = {cam_name: exptimes_list * u.second for cam_name in camera_list}
+
+        # Loop until conditions are met for dark fields
+        while True:
+            self.status()  # Seems to help with reading coords
+
+            fits_headers = self.get_standard_headers(observation=dark_obs)
+
+            start_time = utils.current_time()
+            fits_headers['start_time'] = utils.flatten_time(
+                start_time)  # Common start time for cameras
+
+            camera_events = dict()
+
+            # Take the observations
+            for cam_name in camera_list:
+
+                camera = self.cameras[cam_name]
+
+                filename = "{}/darks/{}/{}/{}.{}".format(
+                    image_dir,
+                    camera.uid,
+                    dark_obs.seq_time,
+                    'dark_{:02d}'.format(dark_obs.current_exp_num),
+                    camera.file_extension)
+
+                # Take picture and get event
+                camera_event = camera.take_observation(
+                    dark_obs,
+                    fits_headers,
+                    filename=filename,
+                    exptime=exptimes[cam_name][-1]
```

Comment:
Should this not be a loop over exposure times? Why is `exptimes[cam_name][-1]` used?

Reply:
The [-1] was a typing error. Thanks for this!